API/API/Controllers/ProductController.cs

```diff
@@ -1,109 +1,96 @@
+            return BadRequest(ex.Message);
         }
+    }
 
-        /// <summary>
-        /// Post product data
-        /// </summary>
-        [HttpPost]
-        public async Task<ActionResult<ProductItem>> PostProductItem(ProductItem productItem)
+    /// <summary>
+    /// Post product data
+    /// </summary>
+    [HttpPost]
+    public async Task<ActionResult<ProductModel>> PostProduct(ProductModel PostProduct)
+    {
+        try
         {
-
-            _context.ProductItems.Add(productItem);
-            await _context.SaveChangesAsync();
-
-            return CreatedAtAction(nameof(GetProductItem), new { id = productItem.Id }, productItem);
+            await _productService.CreateProduct(PostProduct);
+            return Ok(PostProduct);
         }
-
-        /// <summary>
-        /// Put product data
-        /// </summary>
-        [HttpPut]
-        public async Task<IActionResult> PutProductItem(int id,ProductItem productItem)
+        catch (ExceptionClass ex)
         {
-            if (id != productItem.Id)
-            {
-                return BadRequest();    
-            }
-
-            _context.Entry(productItem).State= EntityState.Modified;
+            return BadRequest(ex.Message);
+        }
+    }
 
-            try
-            {
-                await _context.SaveChangesAsync();
-            }
-            catch(DbUpdateConcurrencyException)
-            {
-                if (!ProductExists(id))
-                {
-                    return NotFound();
-                }
-                else
-                {
-                    throw;
-                }
-            }
-            //return NoContent();
-            return CreatedAtAction(nameof(GetProductItem), new { id = productItem.Id }, productItem);
+    /// <summary>
+    /// Put product data
+    /// </summary>
+    [HttpPut]
+    public async Task<ActionResult<ProductModel>> PutProduct(ProductModel PutProduct)
+    {
+        try
+        {
 
+            return Ok(await _productService.UpdateProduct(PutProduct));
         }
-
-        
-        private bool ProductExists(int id)
+        catch (ExceptionClass ex)
         {
-            return _context.ProductItems.Any(x => x.Id == id);
+            //return BadRequest(ex.Message);
+            return BadRequest("Product does not exist, you can´t modify");
         }
+    }
 
-        /// <summary>
-        /// Delete product data
-        /// </summary>
-        [HttpDelete("{id}")]
-        public async Task<IActionResult> DeleteProductItem(int id)
-        {
-            var productItem = await _context.ProductItems.FindAsync(id);
-            if (productItem == null)
-            {
-                return NotFound();
-            }
-            _context.ProductItems.Remove(productItem);
-            await _context.SaveChangesAsync();
 
-            return NoContent();
+    /// <summary>
+    /// Delete product data
+    /// </summary>
+    [HttpDelete("{DeleteProduct}")]
```

Comment:
The comment is not complete you shoud add parameters and returns

Reply:
Done

Reply:
I don´t add a return because Delete doesn´t return anything